src/main/java/nextstep/subway/favorite/application/FavoriteService.java

```diff
@@ -0,0 +1,71 @@
+package nextstep.subway.favorite.application;
+
+import nextstep.subway.auth.domain.LoginMember;
+import nextstep.subway.common.ErrorMessage;
+import nextstep.subway.common.exception.DuplicationException;
+import nextstep.subway.common.exception.NotFoundException;
+import nextstep.subway.favorite.domain.Favorite;
+import nextstep.subway.favorite.domain.FavoriteRepository;
+import nextstep.subway.favorite.dto.FavoriteRequest;
+import nextstep.subway.favorite.dto.FavoriteResponse;
+import nextstep.subway.member.application.MemberService;
+import nextstep.subway.member.domain.Member;
+import nextstep.subway.station.application.StationService;
+import nextstep.subway.station.domain.Station;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Transactional
+@Service
+public class FavoriteService {
+    private final FavoriteRepository favoriteRepository;
+    private final StationService stationService;
+    private final MemberService memberService;
+
+    public FavoriteService(FavoriteRepository favoriteRepository, StationService stationService, MemberService memberService) {
+        this.favoriteRepository = favoriteRepository;
+        this.stationService = stationService;
+        this.memberService = memberService;
+    }
+
+    public FavoriteResponse register(LoginMember loginMember, FavoriteRequest request) {
+        Member member = memberService.findById(loginMember.getId());
+        Station source = stationService.findById(request.getSourceId());
+        Station target = stationService.findById(request.getTargetId());
+
+        validateDuplication(member, source, target);
+
+        Favorite favorite = new Favorite(member, source, target);
+        favoriteRepository.save(favorite);
+        return FavoriteResponse.of(favorite);
+    }
+
+    @Transactional(readOnly = true)
+    public List<FavoriteResponse> findAll(LoginMember loginMember) {
+        Member member = memberService.findById(loginMember.getId());
+        List<Favorite> favorites = favoriteRepository.findAllByMember(member);
```

Comment:
memberId로 조회하면 매번 Member를 조회하지 않아도 될 것 같은데 어떻게 생각하시나요?

Reply:
흠..이부분은 잘 이해가 안갑니다ㅜㅜ

Reply:
```suggestion
        List<Favorite> favorites = favoriteRepository.findAllByMemberId(loginMember.getId());
```

와 같이 개선해볼 수 있을 것 같아요 ㅎㅎ